sirepo/package_data/static/js/sirepo-beamline.js

```diff
@@ -843,6 +843,9 @@ SIREPO.app.directive('beamlineAnimation', function(appState, frameCache, persist
         scope: {},
         template: `
           <div class="col-sm-3">
+            <div data-canceled-due-to-timeout-alert="simState"></div>
+            <div data-simulation-stopped-status="simState" data-ng-if="simState.stateAsText() != 'Missing' "></div>
```

Comment:
I think we should add "isStateMissing" and check the string in there. I'm also not convinced we need it. If the state is missing then getCanceledAfterSecs should be falsey which would mean the showAlert check in simulationStoppedStatus is false. I haven't tested this code so I could be missing something.

In addition do we need to add a check for isStopped?

Reply:
I think that you were right about `simState.stateAsText() != 'Missing'` being redundant. Do you want me to add a isStateMissing to persistentSimulation?

Also, not sure what you mean by adding a check for isStopped? Do you mean adding another state checking function to persistentSimulation?

Reply:
No need to add isStateMissing since we don't need it.

I was curious if we need to check for isStopped like [we do in elegant](https://github.com/radiasoft/sirepo/blob/1a3676c3aeb588c97f6129520eb7a70d90128f44/sirepo/package_data/static/html/elegant-visualization.html#L25)

Reply:
Sorry I'm not being clear - I'm wondering if we need to protect the call to `simulation-stopped-status` like we do in other places. Ex:
- guarded with frameCount > 0 https://github.com/radiasoft/sirepo/blob/be9e4bcae98faeb284428421b450eb42450ed10a/sirepo/package_data/static/js/sirepo-components.js#L4389
- guarded with isStopped https://github.com/radiasoft/sirepo/blob/be9e4bcae98faeb284428421b450eb42450ed10a/sirepo/package_data/static/js/srw.js#L2542
- guarded with isStopped https://github.com/radiasoft/sirepo/blob/be9e4bcae98faeb284428421b450eb42450ed10a/sirepo/package_data/static/html/elegant-visualization.html#L23